### PR TITLE
EF Template Compatibility

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this?
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

ported and adapted templates from community #3441
also port of #3442

### Please specify which Issue this PR Resolves (If Applicable).
"This PR closes #XXXX!"

### Please verify the following.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:

### CODE REVIEW

**Comment by wersal454** [comment]
Also, don't forget about doing "prices" in Vehicle_Attributes

**Comment by wersal454** [comment]
Untested

Need to launch every template to check if there are any errors
Also need to check cargo config for the new turret

File: A3A/addons/logistics/Nodes/EF.hpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+class ef_ef_hunter_ef_mrap_01_fsv : TRIPLES(ADDON,Nodes,Base)
+{
+    class Nodes
+    {
+        class Node1
+        {
+            offset[] = {-0.47,-3.4,-1.77};
+        };
+    };
+};
```
Comment (Silence-112) [inline A3A/addons/logistics/Nodes/EF.hpp]: This is already defined
```suggestion
```

**Review by Silence-112 — CHANGES_REQUESTED** [review]

File: A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Arid.sqf [hunk 2]
```diff
@@ -0,0 +1,1442 @@
...
+["vehiclesLightAPCs", _lightAPCs] call _fnc_saveToTemplate;
+["vehiclesAPCs", _APCs] call _fnc_saveToTemplate;
+["vehiclesIFVs", _IFVs] call _fnc_saveToTemplate;
+["vehiclesLightUnarmed", _unarmedVehicles] call _fnc_saveToTemplate;
+["vehiclesLightArmed", _armedVehicles] call _fnc_saveToTemplate;
+["vehiclesLightTanks",  _lightTanks] call _fnc_saveToTemplate;
+["vehiclesAirborne", _airborneVehicles] call _fnc_saveToTemplate;
+["vehiclesMilitiaAPCs", _militiaAPCs] call _fnc_saveToTemplate;
+
+#include "Vanilla_Vehicle_Attributes.sqf"
```
Comment (Silence-112) [inline A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Arid.sqf]:
```suggestion
#include "..\Vanilla\Vanilla_Vehicle_Attributes.sqf"
```

File: A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Temperate.sqf [hunk 3]
```diff
@@ -0,0 +1,1439 @@
...
+["vehiclesLightAPCs", _lightAPCs] call _fnc_saveToTemplate;
+["vehiclesAPCs", _APCs] call _fnc_saveToTemplate;
+["vehiclesIFVs", _IFVs] call _fnc_saveToTemplate;
+["vehiclesLightUnarmed", _unarmedVehicles] call _fnc_saveToTemplate;
+["vehiclesLightArmed", _armedVehicles] call _fnc_saveToTemplate;
+["vehiclesLightTanks",  _lightTanks] call _fnc_saveToTemplate;
+["vehiclesAirborne", _airborneVehicles] call _fnc_saveToTemplate;
+["vehiclesMilitiaAPCs", _militiaAPCs] call _fnc_saveToTemplate;
+
+#include "Vanilla_Vehicle_Attributes.sqf"
```
Comment (Silence-112) [inline A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Temperate.sqf]: This path doesn't exist
```suggestion
#include "..\Vanilla\Vanilla_Vehicle_Attributes.sqf"
```

File: A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Arid.sqf [hunk 4]
```diff
@@ -0,0 +1,1442 @@
...
+["vehiclesRepairTrucks", _repairTrucks] call _fnc_saveToTemplate;
+["vehiclesFuelTrucks", _fuelTrucks] call _fnc_saveToTemplate;
+["vehiclesMedical", _medicalTrucks] call _fnc_saveToTemplate;
+["vehiclesBasic", _basic] call _fnc_saveToTemplate;
+["vehiclesTanks", _tanks] call _fnc_saveToTemplate;
+["uavsPortable", _uavsPortable] call _fnc_saveToTemplate;
+
+["vehiclesHelisTransport", _transportHelicopters] call _fnc_saveToTemplate;
+["vehiclesPolice", _policeVehs] call _fnc_saveToTemplate;
+["vehiclesHelisLightAttack", _helisLightAttack] call _fnc_saveToTemplate;
```
Comment (Silence-112) [inline A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Arid.sqf]: These variables did not exist
```suggestion
["vehiclesPolice", _vehiclesPolice] call _fnc_saveToTemplate;
["vehiclesHelisLightAttack", _vehiclesHelisLightAttack] call _fnc_saveToTemplate;
```

File: A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Arid.sqf [hunk 5]
```diff
@@ -0,0 +1,1442 @@
...
+["vehiclesPlanesTransport", _planesTransport] call _fnc_saveToTemplate;
+["vehiclesHelisLight", _helisLight] call _fnc_saveToTemplate;
+["vehiclesHelisAttack", _helisAttack] call _fnc_saveToTemplate;
```
Comment (Silence-112) [inline A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Arid.sqf]:
```suggestion
["vehiclesHelisLight", _vehiclesHelisLight] call _fnc_saveToTemplate;
["vehiclesHelisAttack", _vehiclesHelisAttack] call _fnc_saveToTemplate;
```

File: A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Temperate.sqf [hunk 6]
```diff
@@ -0,0 +1,1439 @@
...
+["vehiclesAmmoTrucks", _ammoTrucks] call _fnc_saveToTemplate;
+["vehiclesRepairTrucks", _repairTrucks] call _fnc_saveToTemplate;
+["vehiclesFuelTrucks", _fuelTrucks] call _fnc_saveToTemplate;
+["vehiclesMedical", _medicalTrucks] call _fnc_saveToTemplate;
+["vehiclesBasic", _basic] call _fnc_saveToTemplate;
+["vehiclesTanks", _tanks] call _fnc_saveToTemplate;
+["uavsPortable", _uavsPortable] call _fnc_saveToTemplate;
+["vehiclesHelisTransport", _transportHelicopters] call _fnc_saveToTemplate;
+["vehiclesPolice", _policeVehs] call _fnc_saveToTemplate;
+["vehiclesHelisLightAttack", _helisLightAttack] call _fnc_saveToTemplate;
```
Comment (Silence-112) [inline A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Temperate.sqf]:
```suggestion
["vehiclesPolice", _vehiclesPolice] call _fnc_saveToTemplate;
["vehiclesHelisLightAttack", _vehiclesHelisLightAttack] call _fnc_saveToTemplate;
```

File: A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Temperate.sqf [hunk 7]
```diff
@@ -0,0 +1,1439 @@
...
+["vehiclesHelisLight", _helisLight] call _fnc_saveToTemplate;
+["vehiclesHelisAttack", _helisAttack] call _fnc_saveToTemplate;
```
Comment (Silence-112) [inline A3A/addons/core/Templates/Templates/EF/EF_AI_MJTF_Temperate.sqf]:
```suggestion
["vehiclesHelisLight", _vehiclesHelisLight] call _fnc_saveToTemplate;
["vehiclesHelisAttack", _vehiclesHelisAttack] call _fnc_saveToTemplate;
```

**Review by Silence-112 — CHANGES_REQUESTED** [review]
Double check the Vanilla_AI_XYZ templates as quite a few are missing the `private _hasEF = "ef" in A3A_enabledDLC;` declaration

**Review by Silence-112 — CHANGES_REQUESTED** [review]
The 3 `Vanilla_AI_NATO_Side` templates are missing the `_gunBoat` variable used in DLC_content includes, should double check for any other variables missed

**Review by Silence-112 — APPROVED** [review]
Lgtm

**Pull request merged**